### PR TITLE
feat(builtin.buffers): enhance and bind `delete_buffer` action

### PR DESCRIPTION
**Pull request by jamestrew**
Binds the `delete_buffer` action to `<C-x>` by default for the `buffers` pickers.

closes https://github.com/nvim-telescope/telescope.nvim/issues/3128

Also enhances the `delete_buffer` action to have a "cleaner" behavior for deleting the current buffer (the buffer under telescope). Previously, when deleting the current buffer, the buffer would be removed from the buflist but closing the picker would put you right back into the buffer you just deleted.
From `:h bdelete`:
> If buffer [N] is the current buffer, another buffer will be displayed instead.  This is the most recent entry in the jump list that points into a loaded buffer.

Since the telescope prompt buffer is technically the current buffer, this "another buffer will be displayed behavior" wasn't actually triggering.
With this newest change, Telescope will manually get the buffer of the most recent entry in the jumplist and change the window behind telescope to this new buffer.

closes https://github.com/nvim-telescope/telescope.nvim/issues/2643

### CODE REVIEW

File: lua/telescope/builtin/__internal.lua [hunk 1]
```diff
@@ -972,6 +972,9 @@ internal.buffers = function(opts)
       previewer = conf.grep_previewer(opts),
       sorter = conf.generic_sorter(opts),
       default_selection_index = default_selection_idx,
+      attach_mappings = function(_, map)
+        map({ "i", "n" }, "<C-x>", actions.delete_buffer)
+      end,
```
Comment (Conni2461) [inline lua/telescope/builtin/__internal.lua:977]: I am not the biggest fan of mapping custom picker actions by default, thats why its currently still not mapped. Mappings are always opinionation but we can do that but should document that at least in `builtins/init.lua`, with a simple mention

**Review by Conni2461 — APPROVED** [review]
LGTM, besides docs nit

**Comment by Conni2461** [comment]
Thanks :)

**Comment by jamestrew** [comment]
Thanks, I agree about setting default mappings but it's such a common question I see here/reddit/discord so.
Switched it to `<M-d>` since I forgot that `<C-x>` is already a default mapping.

**Comment by igorlfs** [comment]
Hey! This commit introduced a bug :(

I can no longer use `:Telescope buffers`. I get a stacktrace that points to 
```telescope.nvim/lua/telescope/mappings.lua:330: Attach mappings must always return a value. `true` means use default mappings, `false` means only use attached mappings```. So I guess this should be retuning true?

**Pull request merged**